l2geth/rollup/fees/rollup_fee.go

```diff
@@ -87,6 +99,54 @@ func DecodeL2GasLimitU64(gasLimit uint64) uint64 {
 	return scaled * tenThousand
 }
 
+// PaysEnoughOpts represent the options to PaysEnough
+type PaysEnoughOpts struct {
+	GasLimit                   uint64
+	GasPrice, Fee              *big.Int
+	ThresholdUp, ThresholdDown *big.Float
+}
+
+// PaysEnough returns an error if the fee is not large enough
+// `GasPrice` and `Fee` are required arguments.
+func PaysEnough(opts *PaysEnoughOpts) error {
```

Comment:
Would be good to add more input validation -- eg. `opts.ThresholdUp >= 1` & `opts.ThresholdDown <= 1 && >= 0`

Reply:
This validation happens in the constructor of the sync service

Reply:
Hmm I guess I slightly prefer co-locating the input validation with the struct declaration as much as possible

Reply:
The problem is a `NewPaysEnoughOpts` function would revert to having the problem that I was trying to avoid which was having a bunch of ordered arguments, the point of `PaysEnoughOpts` was that it would make it very explicit what the args were. We could add the checking of the threshold up and down in this function